Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -42,6 +42,13 @@ def edit
       EOS
     end
 
+    unless Homebrew::EnvConfig.no_env_hints?
+      opoo <<~EOS
+        Unless `HOMEBREW_NO_INSTALL_FROM_API` is set when running
+        `brew install`, it will ignore your locally edited formula.
```

Comment:
```suggestion
        For your changes to take effect, you must set `HOMEBREW_NO_INSTALL_FROM_API`.
```

Reply:
I'd mention this only applies to core formulae/casks too.

Ideally this message would also only display if you pass an arg to a core formula/cask and not when you edit a tap one.

Reply:
@Bo98 Done in [79beebe](https://github.com/Homebrew/brew/pull/14570/commits/79beebe99d13afc350e995aeaca327fcde63c2a4) (first time writing ruby I hope it's correct 👀)

@carlocab I think `brew install` should be mentioned explicitly since it's otherwise unclear/potentially misleading since this is printed when you run `brew edit` (but again please feel free to edit directly)